doc/bsp/raspberrypi.md

```diff
@@ -0,0 +1,330 @@
+$ sudo mkfs.ext4 -L rootfs /dev/sde2
+mke2fs 1.44.6 (5-Mar-2019)
+/dev/sde2 contains a ext4 file system labelled 'rootfs'
+        created on Thu Aug  8 09:43:13 2019
+Proceed anyway? (y,N) y
+Creating filesystem with 245504 4k blocks and 61440 inodes
+Filesystem UUID: 7ce1bfb2-4862-4dea-82a3-e5b782115d7b
+Superblock backups stored on blocks: 
+        32768, 98304, 163840, 229376
+
+Allocating group tables: done
+Writing inode tables: done
+Creating journal (4096 blocks): done
+Writing superblocks and filesystem accounting information: done
+```
+_note: command line adds rootfs label_
+
+5. Mount file systems
+
+```
+$ sudo mount /dev/sde1 /mnt/rpi
+$ sudo mount /dev/sde2 /mnt/rootfs
+```
+
+## Copy Raspberry Pi firmwares
+
+1. Clone firmware repository
+
+```
+$ git clone git@github.com:raspberrypi/firmware.git
+```
+
+2. Copy files to fat32 partition
+
+```
+$ cd firmware
+$ cp -a boot/start* /mnt/rpi/.
+$ cp -a boot/fixup* /mnt/rpi/.
+$ cp -a boot/overlays/ /mnt/rpi/.
+```
+
+## Copy kernel and dtb 
+
+Go to your build directory.
+
+```
+$ cp tmp-glibc/deploy/images/raspberrypi3-64/Image /mnt/rpi/.
+$ cp tmp-glibc/deploy/images/raspberrypi3-64/bcm2837-rpi-3-b-plus.dtb /mnt/rpi/.
+```
+
+##  Create config.txt
+
+1. Create config.txt on /mnt/rpi
+
+```
+[all]
+boot_delay=1
+kernel=u-boot.bin
+
+# Put the RPi3 into 64 bit mode
+arm_control=0x200
+
+upstream_kernel=1
+audio_pwm_mode=0
+
+# Enable UART
+enable_uart=1
+
+gpu_mem=16
+mask_gpu_interrupt1=0x100
+```
+
+## Setup u-boot
+
+1. Copy u-boot binary
+
+```
+$ cd u-boot
+$ cp ./u-boot.bin /mnt/rpi/.
+```
+
+2. Create boot.cmd file on any directory.
+
+```
+fatload mmc 0 ${kernel_addr_r} Image
+fatload mmc 0 ${fdt_addr_r} bcm2837-rpi-3-b-plus.dtb
+setenv bootargs dwc_otg.lpm_enable=0 earlyprintk root=/dev/mmcblk0p2 rootfstype=ext4 rootwait 
+booti ${kernel_addr_r} - ${fdt_addr_r}
+```
+
+3. Create .src file
+
+```
+$ sudo mkimage -C none -A arm64 -T script -d ./boot.cmd /mnt/rpi/boot.scr
+Image Name:
+Created:      Thu Aug  8 11:46:12 2019
+Image Type:   AArch64 Linux Script (uncompressed)
+Data Size:    256 Bytes = 0.25 KiB = 0.00 MiB
+Load Address: 00000000
+Entry Point:  00000000
+Contents:
+   Image 0: 248 Bytes = 0.24 KiB = 0.00 MiB
+```
+
+4. (optional) Enable early UART support
+
+If you want see u-boot's message by uart, you needs to run following command.
+
+```
+$ sudo sed -i -e "s/BOOT_UART=0/BOOT_UART=1/" /mnt/rpi/bootcode.bin 
+```
+
+## Setup root file system
+
+1. Extract root file system files to sdcard
+
+Go to build directory then.
+
+```
+$ sudo tar xvf tmp-glibc/deploy/images/raspberrypi3-64/core-image-minimal-raspberrypi3-64.tar.gz -C /mnt/rootfs/
+```
+
+2. Clone linux firmwares from git repository 
+
+```
+$ git://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git
+```
+
+3. Create /lib/firmware on rootfs partition
+
+```
+$ sudo mkdir  /mnt/rootfs/lib/firmware
+```
+
+4. Copy firmwares
+
+```
+$ sudo cp -a linux-firmware/brcm/ /mnt/rootfs/lib/firmware/
```

Comment:
`cp -r`. Otherwise, the files are owned by non-root user.

Reply:
fixed. thanks!